packages/data/config/navi-config.d.ts

```diff
@@ -7,8 +7,14 @@ declare module 'navi-config' {
     description?: string;
     uri: string;
     type: string;
+    options?: FiliConfigOptions | ElideConfigOptions;
```

Comment:
Can we do something like

```ts
type BaseDataSource<Type, Options> = {
  name: string;
  displayName: string;
  description?: string;
  uri: string;
  type: Type;
  options?: Options;
};
export interface FiliConfigOptions {
  enableDimensionSearch?: boolean;
}
export interface ElideConfigOptions {
  // elide options here
}

type NaviDataSource = BaseDataSource<'bard', FiliConfigOptions> | BaseDataSource<'elide', ElideConfigOptions>
```

So that only the options for the current dataSource can be used